src/appleseed/foundation/utility/poison.h

```diff
@@ -65,7 +65,10 @@ namespace foundation
 //
 
 template <typename T>
-void poison(T& x);
+void debug_poison(T& x);
```

Comment:
The comment above needs updating both regarding which free function to use when, and how to implement `do_poison()`.

Reply:
Ah, sorry. On it.